eng/devices/android.cake

```diff
@@ -213,8 +213,9 @@ Task("Test")
 	Information("Test App Package Name: {0}", TEST_APP_PACKAGE_NAME);
 	Information("Test App Instrumentation: {0}", TEST_APP_INSTRUMENTATION);
 	Information("Test Results Directory: {0}", TEST_RESULTS);
-
-	CleanDirectories(TEST_RESULTS);
+	
+	if (IsCIBuild())
+		CleanDirectories(TEST_RESULTS);
```

Comment:
Is this ok @rmarinho? Will there be results left over from other CI runs that we need to clear out?